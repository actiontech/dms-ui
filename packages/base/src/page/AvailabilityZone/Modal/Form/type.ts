import { FormInstance } from 'antd';

export type AvailabilityZoneFormType = {
  name: string;
  address: string;
};

export type AvailabilityZoneFormProps = {
  form: FormInstance<AvailabilityZoneFormType>;
  isUpdate?: boolean;
};
