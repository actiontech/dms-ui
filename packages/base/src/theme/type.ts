import { WorkflowRecordStatusEnum } from '@actiontech/shared/lib/api/base/service/common.enum';

export type SideMenuTheme = {
  suffixIconColor: string;
  projectSelector: {
    dropdown: {
      labelColor: string;
      iconColor: string;
      activeIconColor: string;
      groupLabelColor: string;
    };
  };
};

export type GuidanceTheme = {
  padding: number;
  gap: number;
  sceneSegmented: {
    backgroundColor: string;
    borderBottom: string;
  };
  steps: {
    icon: {
      width: number;
      height: number;
      backgroundColor: string;
    };
    connection: {
      width: number;
      gap: number;
      backgroundColor: string;
    };
    step: {
      border: string;
      title: {
        borderRight: string;
        index: {
          color: string;
        };
        text: {
          color: string;
        };
      };
      content: {
        desc: {
          color: string;
        };
      };
    };
  };
  guidanceButton: {
    default: {
      background: string;
      color: string;
    };
    hover: {
      background: string;
      color: string;
    };
  };
};

export type SystemTheme = {
  configButton: {
    backgroundColor: string;
    border: string;
  };
  logo: {
    boxShadow: string;
  };
};

export type DataExportTheme = {
  create: {
    form: {
      baseInfoTitleIconColor: string;
    };
  };
  statistics: {
    auditResultStatusColor: {
      [key in WorkflowRecordStatusEnum]: string;
    };
  };
};

export type IconTheme = {
  bindUser: {
    user: string;
    password: string;
  };
  dataExport: {
    infoCircle: string;
    clock: string;
  };
  home: {
    common: string;
  };
  system: {
    basicTitleTips: string;
    modify: string;
  };
};
