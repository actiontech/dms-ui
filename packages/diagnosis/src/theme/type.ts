export type MonitorSourceConfigTheme = {
  headerWrapper: {
    borderBottom: string;
    title: {
      fontSize: string;
      fontWeight: number;
      color: string;
    };
    refreshIcon: {
      color: string;
    };
    tagWrapper: {
      fontSize: string;
      fontWeight: number;
      color: string;

      tagItem: {
        border: string;
        backgroundColor: string;
      };
      tagPrimary: {
        primaryColor: string;
        hoverColor: string;
      };
      icon: {
        backgroundColor: string;
      };
    };
  };
};

export type LoginTheme = {
  logo: {
    boxShadow: string;
  };
  icon: {
    user: string;
    password: string;
  };
};

export type UserManagementTheme = {
  permissionType: {
    border: string;
    backgroundColor: string;
    color: string;
    activeBackgroundColor: string;
    activeColor: string;
  };
};
