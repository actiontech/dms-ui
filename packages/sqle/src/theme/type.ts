import { WorkflowRecordResV2StatusEnum } from '@actiontech/shared/lib/api/sqle/service/common.enum';

export type ExecWorkflowTheme = {
  list: {
    desc: {
      copyIconColor: string;
      hoverCopyIconBackgroundColor: string;
    };
  };
  create: {
    auditResult: {
      auditResultDrawer: {
        numberColor: string;
      };
      toggleButton: {
        borderColor: string;
        color: string;
        activeColor: string;
        bgColor: string;
        activeBgColor: string;
      };
      download: {
        borderColor: string;
        bgColor: string;
        boxShadow: string;
        itemColor: string;
        itemHoverColor: string;
        itemIconColor: string;
      };
    };
    form: {
      baseInfoTitleIconColor: string;
    };
    editForm: {
      titleColor: string;
      projectFlagIconColor: string;
    };
  };
  steps: {
    boxShadow: string;
  };
  common: {
    basicInfo: {
      borderColor: string;
      bgColor: string;
      titleColor: string;
      descColor: string;
    };
    auditResultFilter: {
      borderColor: string;
      bgColor: string;
      auditResultInfo: {
        itemBgColor: string;
        textColor: string;
        schemaValueColor: string;
      };
      options: {
        bgColor: string;
        activeBgColor: string;

        textColor: string;
        textActiveColor: string;
        numColor: string;
        numActiveColor: string;
      };
    };
  };
};

export type ReportStatisticsTheme = {
  bgColor: string;
  loadingColor: string;
  cardShow: {
    titleColor: string;
    iconColor: string;
    numberContColor: string;
    noteColor: string;
    contentFontSize: number;
    contentFontWeight: number;
  };
  CardWrapper: {
    titleColor: string;
    titleBorderColor: string;
  };
  TableTopList: {
    titleColor: string;
    noteTipColor: string;
    colorFillQuaternary: string;
    bgColor: {
      toColor: string;
      line1: string;
      line2: string;
      line3: string;
    };
  };
  ChartContTitle: {
    mainColor: string;
    mainSubColor: string;
    subContColor: string;
  };
  DatabaseTypeOrder: {
    default: {
      fontSize: number;
    };
    tooltip: {
      background: string;
      border: string;
      boxShadow: string;
    };
  };
  DatabaseSourceOrder: {
    default: {
      colorText: string;
      colorTextTertiary: string;
    };
  };
  LicenseStatistics: {
    ChartContTitle: {
      lineColor: string;
    };
    LicenseColumn: {
      defaultColor: string;
      fillColor: {
        column: string;
        columnBackground: string;
        xAxis: string;
        yAxis: string;
        state: {
          active: string;
        };
      };
    };
  };
  WorkOrderState: {
    ChartContTitle: {
      color: string;
      noteSubColor: string;
    };
  };
};

export type StatisticsTheme = {
  rectColor: string[];
  rectColorName: { [key: string]: string };
  auditRateStatus: {
    success: {
      color: string;
      bg: string;
    };
    warning: {
      color: string;
      bg: string;
    };
    error: {
      color: string;
      bg: string;
    };
    tip: {
      color: string;
      bg: string;
    };
  };
  auditResultStatusColor: {
    [key in WorkflowRecordResV2StatusEnum]: string;
  };
};

export type ProjectOverviewTheme = {
  SqlCount: {
    baseColor: string;
    grayColor: string;
  };
  DataSourceCount: {
    health: string;
    risk: string;
  };
  ProjectScore: {
    level: {
      dangerous: string;
      warning: string;
      good: string;
      excellent: string;
    };
    indicator: {
      pointer: string;
      pin: {
        fill: string;
        stroke: string;
      };
    };
  };
  ScanTask: {
    bar: {
      fill: string;
      bg: string;
      activeColor: string;
      label: {
        fill: string;
      };
      toolTip: {
        dotColor: string;
      };
    };
    detail: {
      stroke: string;
    };
  };
  DataSourcePerformance: {
    bar: {
      fill: string;
      bg: string;
    };
    toolTip: {
      dotColor: string;
    };
  };
};

export type WorkflowTemplateTheme = {
  progress: {
    normal: string;
    notice: string;
    warning: string;
    error: string;
    remainColor: string;
  };
  step: {
    honour: string;
    addFile: string;
    audit: string;
    sendPlane: string;
  };
  stepCard: {
    userAvatarBorder: string;
    boxShadow: string;
    hoverBoxShadow: string;
    backgroundColor: string;
    disableBorder: string;
    disabledBackgroundColor: string;
    disableHoverBackgroundColor: string;
    cardDesc: string;
    cardOperator: {
      borderTop: string;
      titleColor: string;
    };
    stepLineColor: string;
    stepReviewIndexTextColor: string;
    closeButtonBoxShadow: string;
    closeButtonFontSize: string;
    cardTitleFontWeight: number;
    descFontSize: string;
    operatorTitleFontWeight: number;
    stepReviewIndexTextFontWeight: number;
    authTextColor: string;
  };
  updateWorkflowTemplateStepInfo: {
    titleFontSize: string;
    titleFontWeight: number;
    infoFontSize: string;
    title: {
      titleWrapperBorderBottom: string;
      titleColor: string;
      titleInfoColor: string;
    };
    info: {
      titleColor: string;
    };
    stepInfoWrapper: {
      border: string;
      backgroundColor: string;
      textColor: string;
    };
    stepNodeAlert: {
      backgroundColor: string;
      alertTitleColor: string;
      alertContentColor: string;
    };
  };
  workflowTemplateAuthInfo: {
    topLevelFontWeight: number;
    authLevelFontWeight: number;
    authLevelFontSize: string;
    authInfoFontSize: string;
    borderBottom: string;
    authInfoColor: string;
  };
};

export type IconTheme = {
  execWorkFlow: {
    minusCircleFilledDisabled: string;
    profileSquareFilled: string;
    databaseFilled: string;
    schemaFilled: string;
    fileList: string;
    clock: string;
  };
  workflowTemplate: {
    common: string;
    userCircleFilled: string;
  };
};

export type DataSourceComparisonTheme = {
  comparisonResultDiffBackgroundColor: string;
  comparisonEntry: {
    card: {
      boxShadow: string;
      borderColor: string;
    };
    timeInfo: {
      color: string;
    };
    overview: {
      card: {
        hoverBoxShadow: string;
        suggestion: {
          color: string;
        };
        mode: {
          same: {
            borderColor: string;
            color: string;
            selectedBgColor: string;
          };
          inconsistent: {
            borderColor: string;
            color: string;
            selectedBgColor: string;
          };
          baseNotExist: {
            borderColor: string;
            color: string;
            selectedBgColor: string;
          };
          comparisonNotExist: {
            borderColor: string;
            color: string;
            selectedBgColor: string;
          };
        };
      };
    };
    treeNode: {
      title: {
        bgColor: string;
      };
      node: {
        inconsistent: {
          color: string;
          baselineBgColor: string;
          comparisonBgColor: string;
        };
        missing: {
          color: string;
          baselineBgColor: string;
          comparisonBgColor: string;
        };
        new: {
          color: string;
          bgColor: string;
        };
        tag: {
          inconsistent: {
            color: string;
            bgColor: string;
          };
          missing: {
            color: string;
            bgColor: string;
          };
          new: {
            color: string;
            bgColor: string;
          };
        };
      };
    };
  };
};

export type KnowledgeTheme = {
  highlight: {
    color: string;
    background: string;
  };
  result: {
    list: {
      boxShadow: string;
      hoverBoxShadow: string;
    };
    description: {
      borderLeftColor: string;
    };
  };
  graph: {
    wrapper: {
      borderColor: string;
      backgroundColor: string;
      dotColor: string;
    };
    container: {
      fullScreenBgColor: string;
    };
    control: {
      borderColor: string;
    };
    edge: {
      color: string;
    };
  };
};
