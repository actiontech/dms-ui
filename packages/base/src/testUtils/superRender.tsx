import {
  superRender,
  superRenderHook,
  WrapperOptions,
  RenderParams
} from '@actiontech/shared/lib/testUtil/superRender';
import { renderHook } from '@testing-library/react-hooks';
import lightTheme from '@actiontech/shared/lib/theme/light';
import basePackageTheme from '../theme/light';

const themeData = {
  ...lightTheme,
  ...basePackageTheme
};

export const baseSuperRender = (
  ...[ui, option, otherProps]: [...RenderParams, WrapperOptions?]
) => {
  return superRender(ui, option, {
    ...otherProps,
    theme: themeData
  });
};

export const baseSuperRenderHook = <TProps, TResult>(
  hook: (props: TProps) => TResult,
  renderHookOptions?: Omit<
    Parameters<typeof renderHook<TProps, TResult>>[1],
    'wrapper'
  >,
  wrapperOptions?: WrapperOptions
) => {
  return superRenderHook(hook, renderHookOptions, {
    ...wrapperOptions,
    theme: themeData
  });
};
