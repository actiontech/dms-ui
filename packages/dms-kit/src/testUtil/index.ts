export * from './superRender';
export * from './customQuery';
export * from './common';
