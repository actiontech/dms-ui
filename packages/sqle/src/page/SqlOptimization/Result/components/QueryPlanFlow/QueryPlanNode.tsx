import React, { useState, useMemo } from 'react';
import { Handle, Position } from '@xyflow/react';
import type { Node, NodeProps } from '@xyflow/react';
import { Space, Typography } from 'antd';
import { DownOutlined, RightOutlined } from '@ant-design/icons';
import { QueryPlanNodeWrapper } from '../style';
import { ExecutionPlanType } from '../../index.type';

interface QueryPlanItem {
  summary: string[];
  children: QueryPlanItem[];
  operator: string;
  [x: string]: any;
}

interface QueryPlanNodeData extends Record<string, unknown> {
  detail: QueryPlanItem;
  level: number;
  hasChildren?: boolean;
  isRootNode?: boolean;
  nodeIndex?: number;
  planType?: ExecutionPlanType;
}

const QueryPlanNode: React.FC<NodeProps<Node<QueryPlanNodeData>>> = ({
  data,
  isConnectable
}) => {
  const {
    detail,
    hasChildren = false,
    isRootNode = false,
    nodeIndex,
    planType
  } = data;
  const { operator } = detail;
  const [isExpanded, setIsExpanded] = useState(true);

  const content = useMemo(() => {
    let conditionString = '';
    if (detail?.conditions) {
      detail?.conditions?.forEach((item: any) => {
        if (typeof item === 'string') {
          conditionString += JSON.stringify(item);
        }
      });
    }

    const summaryString = detail?.summary?.join('\n');

    const contentStr =
      detail?.columns?.join('') ||
      detail?.table ||
      detail?.join_condition ||
      conditionString ||
      summaryString ||
      '';
    return contentStr;
  }, [detail]);

  const handleToggleExpand = (event: React.MouseEvent) => {
    event.stopPropagation();
    event.preventDefault();
    setIsExpanded(!isExpanded);
  };

  return (
    <QueryPlanNodeWrapper planType={planType}>
      {!isRootNode && (
        <Handle
          type="target"
          position={Position.Top}
          isConnectable={isConnectable}
        />
      )}

      <div className="query-plan-node-card">
        <Space direction="vertical" size="small" style={{ width: '100%' }}>
          <div className="operator-row" onClick={handleToggleExpand}>
            <div className="operator-left">
              <span className="collapse-icon">
                {isExpanded ? <DownOutlined /> : <RightOutlined />}
              </span>
              <Typography.Title className="operator-title" level={5}>
                {operator}
              </Typography.Title>
            </div>
            {nodeIndex !== undefined && (
              <span className="node-index">#{nodeIndex}</span>
            )}
          </div>

          {isExpanded && (
            <div className="content-text">
              <Typography.Text>{content}</Typography.Text>
            </div>
          )}
        </Space>
      </div>

      {hasChildren && (
        <Handle
          type="source"
          position={Position.Bottom}
          isConnectable={isConnectable}
        />
      )}
    </QueryPlanNodeWrapper>
  );
};

export default QueryPlanNode;
