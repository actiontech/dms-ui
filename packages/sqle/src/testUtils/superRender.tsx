import {
  superRender,
  superRenderHook,
  WrapperOptions,
  RenderParams
} from '@actiontech/shared/lib/testUtil/superRender';
import sharedTheme from '@actiontech/shared/lib/theme/light';
import lightTheme from '../theme/light';
import { renderHook } from '@testing-library/react-hooks';
import { storeFactory } from './mockRedux';

const themeData = {
  ...sharedTheme,
  ...lightTheme
};

export const sqleSuperRender = (
  ...[ui, option, otherProps]: [...RenderParams, WrapperOptions?]
) => {
  return superRender(ui, option, {
    ...otherProps,
    theme: themeData,
    storeFactory: storeFactory
  });
};

export const sqleSuperRenderHook = <TProps, TResult>(
  hook: (props: TProps) => TResult,
  renderHookOptions?: Omit<
    Parameters<typeof renderHook<TProps, TResult>>[1],
    'wrapper'
  >,
  wrapperOptions?: WrapperOptions
) => {
  return superRenderHook(hook, renderHookOptions, {
    ...wrapperOptions,
    theme: themeData,
    storeFactory: storeFactory
  });
};
