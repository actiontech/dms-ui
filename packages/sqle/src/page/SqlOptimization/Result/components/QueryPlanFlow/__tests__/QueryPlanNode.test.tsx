import { superRender } from '@actiontech/shared/lib/testUtil/superRender';
import QueryPlanNode from '../QueryPlanNode';
import {
  mockUseCurrentProject,
  mockUseCurrentUser
} from '@actiontech/shared/lib/testUtil';
import { mockReactFlow } from '@actiontech/shared/lib/testUtil/mockModule/mockReactFlow';
import { fireEvent, screen, act } from '@testing-library/react';
import { ExecutionPlanType } from '../../../index.type';
import type { Node, NodeProps } from '@xyflow/react';
import { QueryPlanNodeData } from '../index.type';

describe('QueryPlanNode', () => {
  const mockNodeData = {
    detail: {
      operator: 'Table Scan',
      summary: ['Scanning users table', 'Filter: id > 100'],
      children: []
    },
    level: 0,
    hasChildren: false,
    isRootNode: true,
    nodeIndex: 1
  };

  const mockNodeProps = {
    data: mockNodeData,
    isConnectable: true,
    id: 'test-node'
  } as unknown as NodeProps<Node<QueryPlanNodeData>>;

  beforeEach(() => {
    mockUseCurrentProject();
    mockUseCurrentUser();
    mockReactFlow();
    jest.useFakeTimers();
  });

  afterEach(() => {
    jest.useRealTimers();
    jest.clearAllMocks();
  });

  it('should render root node with basic data', () => {
    const { baseElement } = superRender(<QueryPlanNode {...mockNodeProps} />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should render with original plan type', () => {
    const propsWithPlanType = {
      ...mockNodeProps,
      data: {
        ...mockNodeData,
        planType: ExecutionPlanType.ORIGINAL
      }
    };
    const { baseElement } = superRender(
      <QueryPlanNode {...propsWithPlanType} />
    );
    expect(baseElement).toMatchSnapshot();
  });

  it('should render with optimized plan type', () => {
    const propsWithPlanType = {
      ...mockNodeProps,
      data: {
        ...mockNodeData,
        planType: ExecutionPlanType.OPTIMIZED
      }
    };
    const { baseElement } = superRender(
      <QueryPlanNode {...propsWithPlanType} />
    );
    expect(baseElement).toMatchSnapshot();
  });

  it('should toggle expand/collapse when clicked', async () => {
    superRender(<QueryPlanNode {...mockNodeProps} />);

    const operatorRow = screen.getByText('Table Scan').closest('.operator-row');

    expect(screen.getByText(/Scanning users table/)).toBeInTheDocument();

    await act(async () => {
      fireEvent.click(operatorRow!);
      jest.advanceTimersByTime(100);
    });

    expect(screen.queryByText(/Scanning users table/)).not.toBeInTheDocument();

    // Click to expand again
    await act(async () => {
      fireEvent.click(operatorRow!);
      jest.advanceTimersByTime(100);
    });

    expect(screen.getByText(/Scanning users table/)).toBeInTheDocument();
  });

  it('should render without nodeIndex', () => {
    const propsWithoutIndex = {
      ...mockNodeProps,
      data: {
        ...mockNodeData,
        nodeIndex: undefined
      }
    };
    const { baseElement } = superRender(
      <QueryPlanNode {...propsWithoutIndex} />
    );
    expect(baseElement).toMatchSnapshot();
  });

  it('should render with multiline summary data', () => {
    const multilineData = {
      ...mockNodeData,
      detail: {
        operator: 'Complex Join',
        summary: [
          'First line of operation',
          'Second line with details',
          'Third line with more info'
        ],
        children: []
      }
    };
    const propsWithMultilineData = {
      ...mockNodeProps,
      data: multilineData
    };
    const { baseElement } = superRender(
      <QueryPlanNode {...propsWithMultilineData} />
    );
    expect(baseElement).toMatchSnapshot();
  });
});
