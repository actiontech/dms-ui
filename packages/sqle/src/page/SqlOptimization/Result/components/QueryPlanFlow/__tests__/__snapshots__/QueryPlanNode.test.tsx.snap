// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueryPlanNode should render root node with basic data 1`] = `
<body>
  <div>
    <div
      class="css-1wqoq18"
    >
      <div
        class="query-plan-node-card"
      >
        <div
          class="ant-space ant-space-vertical"
          style="width: 100%;"
        >
          <div
            class="ant-space-item"
            style="margin-bottom: 8px;"
          >
            <div
              class="operator-row"
            >
              <div
                class="operator-left"
              >
                <span
                  class="collapse-icon"
                >
                  <span
                    aria-label="down"
                    class="anticon anticon-down"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                      />
                    </svg>
                  </span>
                </span>
                <h5
                  class="ant-typography operator-title"
                >
                  Table Scan
                </h5>
              </div>
              <span
                class="node-index"
              >
                #
                1
              </span>
            </div>
          </div>
          <div
            class="ant-space-item"
          >
            <div
              class="content-text"
            >
              <span
                class="ant-typography"
              >
                Scanning users table
Filter: id &gt; 100
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`QueryPlanNode should render with multiline summary data 1`] = `
<body>
  <div>
    <div
      class="css-1wqoq18"
    >
      <div
        class="query-plan-node-card"
      >
        <div
          class="ant-space ant-space-vertical"
          style="width: 100%;"
        >
          <div
            class="ant-space-item"
            style="margin-bottom: 8px;"
          >
            <div
              class="operator-row"
            >
              <div
                class="operator-left"
              >
                <span
                  class="collapse-icon"
                >
                  <span
                    aria-label="down"
                    class="anticon anticon-down"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                      />
                    </svg>
                  </span>
                </span>
                <h5
                  class="ant-typography operator-title"
                >
                  Complex Join
                </h5>
              </div>
              <span
                class="node-index"
              >
                #
                1
              </span>
            </div>
          </div>
          <div
            class="ant-space-item"
          >
            <div
              class="content-text"
            >
              <span
                class="ant-typography"
              >
                First line of operation
Second line with details
Third line with more info
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`QueryPlanNode should render with optimized plan type 1`] = `
<body>
  <div>
    <div
      class="css-1wqoq18"
    >
      <div
        class="query-plan-node-card"
      >
        <div
          class="ant-space ant-space-vertical"
          style="width: 100%;"
        >
          <div
            class="ant-space-item"
            style="margin-bottom: 8px;"
          >
            <div
              class="operator-row"
            >
              <div
                class="operator-left"
              >
                <span
                  class="collapse-icon"
                >
                  <span
                    aria-label="down"
                    class="anticon anticon-down"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                      />
                    </svg>
                  </span>
                </span>
                <h5
                  class="ant-typography operator-title"
                >
                  Table Scan
                </h5>
              </div>
              <span
                class="node-index"
              >
                #
                1
              </span>
            </div>
          </div>
          <div
            class="ant-space-item"
          >
            <div
              class="content-text"
            >
              <span
                class="ant-typography"
              >
                Scanning users table
Filter: id &gt; 100
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`QueryPlanNode should render with original plan type 1`] = `
<body>
  <div>
    <div
      class="css-1vz5s1g"
    >
      <div
        class="query-plan-node-card"
      >
        <div
          class="ant-space ant-space-vertical"
          style="width: 100%;"
        >
          <div
            class="ant-space-item"
            style="margin-bottom: 8px;"
          >
            <div
              class="operator-row"
            >
              <div
                class="operator-left"
              >
                <span
                  class="collapse-icon"
                >
                  <span
                    aria-label="down"
                    class="anticon anticon-down"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                      />
                    </svg>
                  </span>
                </span>
                <h5
                  class="ant-typography operator-title"
                >
                  Table Scan
                </h5>
              </div>
              <span
                class="node-index"
              >
                #
                1
              </span>
            </div>
          </div>
          <div
            class="ant-space-item"
          >
            <div
              class="content-text"
            >
              <span
                class="ant-typography"
              >
                Scanning users table
Filter: id &gt; 100
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`QueryPlanNode should render without nodeIndex 1`] = `
<body>
  <div>
    <div
      class="css-1wqoq18"
    >
      <div
        class="query-plan-node-card"
      >
        <div
          class="ant-space ant-space-vertical"
          style="width: 100%;"
        >
          <div
            class="ant-space-item"
            style="margin-bottom: 8px;"
          >
            <div
              class="operator-row"
            >
              <div
                class="operator-left"
              >
                <span
                  class="collapse-icon"
                >
                  <span
                    aria-label="down"
                    class="anticon anticon-down"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                      />
                    </svg>
                  </span>
                </span>
                <h5
                  class="ant-typography operator-title"
                >
                  Table Scan
                </h5>
              </div>
            </div>
          </div>
          <div
            class="ant-space-item"
          >
            <div
              class="content-text"
            >
              <span
                class="ant-typography"
              >
                Scanning users table
Filter: id &gt; 100
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
